### DISCUSSION TITLE
Static text field

### DISCUSSION THREAD

**Discussion by wei2go** [verbb/formie#772] [Ideas]
It would be great if we can have such a field which allow us to put in static text (headings, descriptions, dot points, etc) in between fields. Our clients sometimes have very long forms which can be separated into sections, and they often have section explainers (just text or dot points) shown to users.

I found the HTML field is the only way which can do this (or did I miss anything?), but our clients often want to create and edit the forms themselves and they are mostly not technical so they can't write in HTML. It would be great if Formie can support such a feature natively.

Here's what I picture:

- In the form editor screen, the field editor can be a WYSIWYG editor (e.g. Redactor) - in our cases, the toolbar can just be bold, italic, lists, format, link.
- On the front end, it will just print out the HTML from the WYSIWYG.
- I don't think printing this in email is useful, so maybe don't print them in email?

**Comment by engram-design** [comment]
We've had a similar discussion with a rich text editor or markdown (@javangriff) being available on the form builder, but just render static text on the front-end. So far, we've got the HTML field to suit most needs, but it's not as user-friendly as a rich text editor.

I'm undecided if we change the functionality of the HTML field to allow for Rich Text/Markdown/HTML to be entered, or 3 separate fields. My opinion would be the former, as I want to keep fields under wraps as much as we can, so as not to be too overwhelming with choice.

Unfortunately, a WYSIWYG would need to be our own implementation, instead of Redactor, as we can't rely on that always being installed. We're halfway there with that so it won't be massive. And yes, it'll be a "cosmetic" field.

**Reply by wirebone** [reply]
This sounds good however it would be handy to have the option to be able to bring this into an email notification as well. I am thinking about legal terms that they agree to or conditions with links to resources that would be good to notify the user so they have a copy as well.

**Reply by engram-design** [reply]
Yep, I do plan to add options to most (if not all) about whether to include the content of it in an email notification. Just realised it wasn't an issue though! https://github.com/verbb/formie/issues/787

**Reply by wei2go** [reply]
That sounds great. 👍

**Comment by engram-design** [comment]
Added in [4.0.0-beta.5](https://github.com/verbb/formie/releases/tag/4.0.0-beta.5).
